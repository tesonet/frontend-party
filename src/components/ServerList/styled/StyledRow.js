import React from 'react';
import styled from 'styled-components';
import { Row } from 'reactstrap';

const StyledContainer = styled(Row)`
  align-items: center;
  border-bottom: 1px solid ${props => props.theme.listBorderColor};
  color: ${props => props.theme.listTextColor};
  height: 3.5rem;
  transition: background-color .2s;

  &:hover {
    background-color: ${props => props.theme.bgColorTertiary};
  }

  & .col:last-child {
    text-align: right;
  }
`;

const StyledRow = props => (
  <StyledContainer {...props} />
);

export default StyledRow;
