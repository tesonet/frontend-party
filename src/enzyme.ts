import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import fetchMock from "jest-fetch-mock";

declare let global: { fetch: {} };

global.fetch = fetchMock;
configure({ adapter: new Adapter() });
