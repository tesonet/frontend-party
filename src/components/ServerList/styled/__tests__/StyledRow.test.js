import React from 'react';
import { shallow } from 'enzyme';
import { Row } from 'reactstrap';

import StyledRow from '../StyledRow';

describe('StyledRow component', () => {
  it('should render the <Row /> element.', () => {
    const renderedComponent = shallow(<StyledRow />).dive();

    expect(renderedComponent.type()).toEqual(Row);
    expect(renderedComponent).toMatchSnapshot();
  });
});
