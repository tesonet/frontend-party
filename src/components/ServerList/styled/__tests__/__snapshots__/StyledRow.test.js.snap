// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledRow component should render the <Row /> element. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Styled(Row) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "className": "sc-bdVaJa cWJwMy",
      "tag": "div",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "className": "sc-bdVaJa cWJwMy",
        "tag": "div",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
